src/fileioc/fileioc.asm

```diff
@@ -1393,6 +1400,36 @@ ti_ArchiveHasRoom:
 	dec	a
 	ret
 
+;-------------------------------------------------------------------------------
+ti_SetGCBehavior:
+;Set routines to run before and after a garbage collect would be triggered.
+; args:
+;   sp + 3 : pointer to routine to be run before. Set to 0 to use default handler.
+;	sp + 6 : pointer to routine to be run after. Set to 0 to use default handler.
+; return:
+;   None
+	pop	de
```

Comment:
can someone else just double check that this pop order is correct? it seems that it sets up the pointers backwards but maybe I'm just tired.

Reply:
yeah that apparently changed during my last pull lol. Fixed it

Reply:
actually, no. Just realised that was an intentional optimization

Reply:
I'm aware. I don't think it is ever correct; which is why I am asking.

Reply:
see here  https://github.com/beckadamtheinventor/toolchain/blob/a6118329b37020a9790a18bffaf8058e630b3f42/src/fileioc/fileioc.asm#L1424

Reply:
Dude. I am asking you to look at if you are popping the arguments in the correct order. I know how to write code.